packages/cordova-plugin/ios/OSGeolocationError.swift

```diff
@@ -0,0 +1,50 @@
+enum OSGeolocationMethod: String {
+    case getCurrentPosition
+    case watchPosition
+    case clearWatch
+}
+
+enum OSGeolocationError: Error {
+    case locationServicesDisabled
+    case permissionDenied
+    case permissionRestricted
+    case positionUnavailable
+    case inputArgumentsIssue(target: OSGeolocationMethod)
+    case other(_ error: Error)
+
+    func toDictionary() -> [String: String] {
+        [
+            "code": "OS-PLUG-GLOC-\(String(format: "%04d", code))",
+            "description": description
+        ]
+    }
+}
+
+private extension OSGeolocationError {
+    var code: Int {
+        switch self {
+        case .positionUnavailable: 4
+        case .permissionDenied: 5
+        case .locationServicesDisabled: 11
+        case .permissionRestricted: 12
+        case .inputArgumentsIssue(let target):
+            switch target {
+            case .getCurrentPosition: 13
+            case .watchPosition: 14
+            case .clearWatch: 15
+            }
+        case .other: 16
+        }
+    }
+
+    var description: String {
+        switch self {
+        case .positionUnavailable: "There was en error trying to obtain the location."
+        case .permissionDenied: "Location permission request was denied."
+        case .locationServicesDisabled: "Location services are not enabled."
+        case .permissionRestricted: "Application's use of location services was restricted."
+        case .inputArgumentsIssue(let target): "The '\(target.rawValue)' input parameters aren't valid."
+        case .other(let error): "\(error.localizedDescription)"
```

Comment:
I wonder if it makes sense to return this error with the `localizedDescription` in the Cordova plugin. I get returning it in the Capacitor plugin as it's intended for high-level programmers, but does it make sense to return it in the Cordova plugin? How will this be mapped to one of the errors the plugin can return? I'm thinking of a table similar to [the one we have for the Camera plugin](https://success.outsystems.com/documentation/11/integration_with_external_systems/mobile_plugins/camera_plugin/camera_plugin_reference_page/).

Reply:
We can add something before the description, like: "An unexpected error was returned: \(error.localizedDescription)". But we've had Support Cases (like SSLPinning p.e.) where we've had to provide a workaround that simply added the actual error message to the OS ones for troubleshooting. So I think this is quite useful in cases like that.

Reply:
That is a good point. Maybe all we need is to add the prefix that you mentioned.

Reply:
Just noticed that this was code used before but it's not relevant anymore. So I just removed it.